Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth
# Child of Destiny Adaptation

Fixes #898.

**I use [This Terminology](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/wiki/Terminology) here, please check it out.**

## Short (Public) Changelog:
- Adapted Child of Destiny events.

## Adaptation Changelog:
Changes happend in these events:
- `HF.199`and `HF.200`, start events:
  - Child of Destiny events can't fire for **Evil** rulers of **Death God**, **Sargerite**, **Shath'yar** religions because they already have free CBs.
- `HF.215`, accidently drink poison and survive:
  - It talks about cookery, so **Evil** characters can't get it.
- `HF.217`, vision from highgod
  - It talks about religion and gods, so secular and no-gods religions can't get it.
- `HF.228`, encounter wild beast
  - It talks about hunting, so **Evil** characters and characters of **Nature** religion group can't get it.
- `HF.223`, swarmed by admirers
  - It talks about sexual relationships, so `HF.223` event is disabled if the child can't have them (like if she/he is Lich, Satyr, N'raqi etc.)
  - Future admirers must have not bad relationships and be possible to have sexual relationships with the child (no Gnoll x Murloc romances).
- `HF.235`, gain 'posse'
  - Future friends from `HF.235` must have not bad relationships with the child.
- `HF.227`, gain a lover
  - It talks about sexual relationships, so `HF.227` event is disabled if the child can't have them (like if she/he is Lich, Satyr, N'raqi etc.)
  - Future lovers from `HF.227` must have not bad relationships with the child.
- `HF.240`, gain a friend
  - Future friends from `HF.240` must have not bad relationships with the child.
- `HF.290`, goes on an adventure
  - `HF.290` event are limited by standard adventurer limitations: `can_be_adventurer_trigger` and `can_be_attacked_by_adventurer_trigger`.
- `HF.219`, meet warrior
  - A warrior must have not bad relationships with the child.
- `HF.221`, meet wandering wise man
  - A wise man must have not bad relationships with the child.
- `HF.237`, escape from prison
  - `HF.237` event prefers to pick a savior that have not bad relationships with the child.

## Bugfix Changelog:
- Fixed issue where the child could get claims on Priesthood of Elune in `HF.231` event if the child starts around the Kaldorei Empire.

## Localization Changelog
- Adapted localization of `HF.221`, `HF.222` events. Replaced mentions of Latin language and Latin books by Kalimag language and Kalimag phrases.
- Replaced mention of Mithridates in `HF.215` event by Shadra.
- `HF.228` event mentions beasts depending on adapted regional conditions. For example, if you live on Zandalar, this event shall mention a tiger. If you live in Northrend, the event shall mention a bear.
- Purged mention of Alexander the Great from `bloodline_desc_child_of_destiny` localization.

## Developer Changelog:
- Added `can_go_hunting_trigger` since I need to use these conditions in two places, `focus_hunting` and `HF.210`.
- Added `wc_region_triggers.txt` where I moved all the region-related triggers and created a few new ones for the localization of the `HF.228` event.

## How to test:
- Fire (via `event <Event ID>` console command) any of the events mentioned above and see what happens.
- Play as a child, use `event HF.199` to start the Child of Destiny event chain, try to notice something weird in the event chain (play until you're 20 years old).
